contracts/OwnedRegistrar.sol

```diff
@@ -12,29 +22,58 @@ import "./OwnerResolver.sol";
  * removed by an account with the "authoriser" role.
  *
  * An audit of this code is available here: https://hackmd.io/s/SJcPchO57
+ *
+ * The default resolver for users is alterable in this version, but only the
+ * "owner" can reset it.
  */
 contract OwnedRegistrar is RBAC {
     ENS public ens;
-    OwnerResolver public resolver;
+    OldENS public oldENS;
+    address public resolver;
+    bytes32 public baseNode;
     mapping(uint=>mapping(address=>bool)) public registrars; // Maps IANA IDs to authorised accounts
     mapping(bytes32=>uint) public nonces; // Maps namehashes to domain nonces
 
     event RegistrarAdded(uint id, address registrar);
     event RegistrarRemoved(uint id, address registrar);
     event Associate(bytes32 indexed node, bytes32 indexed subnode, address indexed owner);
     event Disassociate(bytes32 indexed node, bytes32 indexed subnode);
+    event ResolverReset(address newResolver);
 
-    constructor(ENS _ens) public {
+    constructor(ENS _ens, address _resolver, OldENS _oldENS, bytes32 _baseNode) public {
         ens = _ens;
-        resolver = new OwnerResolver(_ens);
+        resolver = _resolver;
+        oldENS = _oldENS;
+        baseNode = _baseNode;
         _addRole(msg.sender, "owner");
+
+    }
+
+    /**
+     * @dev Migrate a name from the previous ENSRegistry and update its resolver.
+     * @param labelHash The hash of the label specifying the subnode.
+     */
+
+    function migrate(uint256 labelHash) public onlyRole("owner") {
+        bytes32 node = keccak256(abi.encodePacked(baseNode, bytes32(labelHash)));
+        address owner = oldENS.owner(node);
+        ens.setSubnodeOwner(baseNode, bytes32(labelHash), address(this));
+        ens.setResolver(node, resolver);
```

Comment:
You shouldn't do this for names that are already in the old registry; this will result in unexpectedly resetting some peoples names to point at other addresses, and could result in lost funds.

Instead you should either migrate the name over without changing the resolver - people can later update to the new resolver if they wish - or copy over any records from the old resolver to the new one. The former would be much preferred, as it has a lower chance of accidentally breaking something.

Reply:
You are an excellent teacher!